grunt/robots.js

```diff
@@ -1,32 +1,25 @@
+"use strict";
+
 module.exports = function ( grunt )
 {
-    "use strict";
-
-    grunt.registerTask( "envRobotsIndex", "Set robots.txt to index the site", function ()
-    {
-        grunt.config( "useRobotsIndex", true );
-    } );
-
-    grunt.registerTask( "envRobotsNoIndex", "Set robots.txt to NOT index the site", function ()
-    {
-        grunt.config( "useRobotsIndex", false );
-    } );
+    var env = grunt.config( "envName" );
+    var host = grunt.config( "env" ).host;
 
     grunt.registerTask( "robots", "Write a robots.txt", function ()
     {
-        if( grunt.config( "useRobotsIndex" ) && grunt.config( "Host" ) )
-        {
-            grunt.log.write( "Writing robots.txt for indexing" );
-            grunt.file.write( "build/robots.txt",
-                "User-agent: *\n"
-                + "Sitemap: " + grunt.config( "Host" ) + "/build/sitemap.txt\n" );
-        }
-        else
+        if( env !== "prod" || !host  )
         {
             grunt.log.write( "Writing robots.txt for NO indexing" );
             grunt.file.write( "build/robots.txt",
                 "User-agent: *\n"
                 + "disallow: /\n" );
+            return;
         }
+
+        // TODO: can we determine if there is a sitemap file or not?
```

Comment:
If we disallow crawling, it won't even look at the sitemap file. Otherwise, it's as @jrit says, and we should be tailoring a sitemap file with at least a line pointing to the index file.

However, if you say that there is a sitemap file, and the crawler can't find one, there isn't really a negative repercussion to that, it will still attempt to crawl as much of the site as it can. You can leave this code in. No need to overcomplicate things when there's no real drawback.